pages/content/amp-dev/documentation/guides-and-tutorials/develop/animations/triggering_css_animations.md

```diff
@@ -46,40 +46,107 @@ To keep you pages lean and speedy, AMP has enforced a 50,000 byte CSS limit in t
 
-```
+```html
  <div id="hamburger" tabindex=1 role=button on="tap:hamburger.toggleClass(class='close')">
 
 ```
 The `toggleClass` action can apply to other elements as well and toggle between two classes by adding the `force` attribute.
 
-```
+```html
 <button on="tap:magicBox.toggleClass(class='invisible', force=true),magicBox.toggleClass(class='visible', force=false)">
-    Disappear
+  Disappear
 </button>
 <button on="tap:magicBox.toggleClass(class='visible', force=true),magicBox.toggleClass(class='invisible', force=false)">
-    Reappear
+  Reappear
 </button>
 ```
 
 If you need to remove a class and disallow reapplication, add the `force` attribute with a value of `false`. If you need to addd a class and disallow removal, add `force` with a value of `true`.
 
-## Animate with Numerous Classes
-
-The [`amp-bind`](../../../../documentation/components/reference/amp-bind.md) extension works by defining different states as a JSON object inside [`<amp-state>`](../../../../documentation/components/reference/amp-bind.md#state).
+## Animate with CSS and state
 
-You can define a list of CSS classes within the `<style amp-custom>` tag in the `head` of the document:
+You can add and remove any number of CSS classes with states using [`amp-bind`](../../../../documentation/components/reference/amp-bind.md).
 
+[example preview="top-frame" playground="true"]
+```html
+<head>
+  <script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js"></script>
+  <style amp-custom>
+    div {
+      height: 100px;
+      width: 100px;
+      margin: 1em;
+      background-color: green;
+      margin-left: 100px;
+      transition: 2s;
+    }
+    .visible {
+      opacity: 1;
+    }
+    .invisible {
+      opacity: 0;
+    }
+    .left {
+      transform: translatex(-50px)
+    }
+    .right {
+      transform: translatex(50px)
+    }
+    button {
+      margin-top:  1rem;
+      margin-left: 1rem;
+    }
+  </style>
+</head>
+<body>
+  <amp-state id="magicBox">
+  <script type="application/json">
+    {
+      "visibleBox": {
+        "className": "visible"
+      },
+      "invisibleBox": {
+        "className": "invisible"
+      },
+      "moveLeft": {
+        "className": "left"
+      },
+      "moveRight": {
+        "className": "right"
+      }
+    }
+  </script>
+</amp-state>
+  <div [class]="magicBox[animateBox].className"> </div>
+  <button on="tap:AMP.setState({animateBox: 'invisibleBox'})">
+    Disappear
+  </button>
+  <button on="tap:AMP.setState({animateBox: 'visibleBox'})">
+    Reappear
+  </button>
+  <button on="tap:AMP.setState({animateBox: 'moveLeft'})">
+    Move Left
+  </button>
+  <button on="tap:AMP.setState({animateBox: 'moveRight'})">
+    Move Right
+  </button>
+</body>
 ```
+[/example]
+
+Define multiple class animations by first adding a list of CSS classes within the `<style amp-custom>` tag in the `head` of the document:
+
+```css
 ...
   <style amp-custom>
```

Comment:
Please remove the `<style` tag, since the code block is marked as css.

I also suggest to remove all those `...` at the start and the end of those code blocks.
They confuse the syntax highlighting and I think it is clear without them that there has to be other code before and after.